### ISSUE TITLE
How are we going to merge Project 4

### ISSUE THREAD

**Issue by Nivirx** [cop2805-bkj/cop2805-project3#17]
So everyone shows an equal amount of work we need to decide on which version to turn in (we can change the version later) but we need to merge the changes we don't want first to show participation.

i.e if we decided to us Kelvins we should merge the other two banches first then overwrite them with the final version we want.

This needs to be done soon because it might take a while to do merge clean ups and the project is due today.

**Comment by DDKGM** [comment]
I say lets use your version. My version not Jonathan's is fully ready.

**Comment by DDKGM** [comment]
Or*

**Comment by Nivirx** [comment]
ok then I am going to play around with getting these to merge cleanly and leave a nice "commit trail". wish me luck 🤣

**Comment by LlanerasJ** [comment]
Okay sounds good sorry at work currently!

**Comment by LlanerasJ** [comment]
I almost have my index file system working just a few more line of code I think and it will be ready.

**Comment by Nivirx** [comment]
We talked about this in the PR for this merge operation(very briefly), but to make note of what I did because each branch was based off of a different version of master I will clarify here.

JonsBranch was forked from a old pre-project3 version so I rebased it and made a new branch from it

Golmez-Ver was forked from the project3 release version but because of merge conflicts I had to squash it down to a single commit BUT all the commit history in in that commit...not ideal...but it works.

bclaus-index-impl was also forked from the same point as Golmez-Ver but I made additional forks from that and merged them back in. I had to modify the history of the repo as well and merge to a new branch.

With every branch now conflict free and based on the same version of master I merged all three into master-merge

JonsBranch was first, then Golmez-Ver then bclaus-index-impl

then I refactored each package to clearly separate each version (the fact that it was already pretty much separate helped immensely with eliminating merge conflicts, Thank you!) and then I Tested the resulting application.

That was what got merged into the master branch.

PS: if you haven't noticed, git hub now supports protected branches so I protected the master branch.
Branch protection forces all merges to a protected branch to come from a PR from a different branch and the PR has to be reviewed by at least 1 member with write privileges.
